src/main/scala/com/gu/newsletterlistcleanse/GetCleanseListLambda.scala

```diff
@@ -1,40 +1,42 @@
 package com.gu.newsletterlistcleanse
 
 import com.amazonaws.services.lambda.runtime.Context
-import org.slf4j.{ Logger, LoggerFactory }
-
-/**
- * This is compatible with aws' lambda JSON to POJO conversion.
- * You can test your lambda by sending it the following payload:
- * {"name": "Bob"}
- */
-class GetCleanseListLambdaInput() {
-  var name: String = _
-  def getName(): String = name
-  def setName(theName: String): Unit = name = theName
-}
+import com.gu.identity.model.EmailNewsletters
+import com.gu.newsletterlistcleanse.db.{Campaigns, CampaignsFromDB}
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.beans.BeanProperty
+
+
+case class GetCleanseListLambdaInput(
+  @BeanProperty
+  newslettersToProcess: List[String]
+)
 
 object GetCleanseListLambda {
 
   val logger: Logger = LoggerFactory.getLogger(this.getClass)
+  val campaigns: Campaigns = new CampaignsFromDB()
+  val newsletters: Newsletters = new Newsletters()
 
-  /*
-   * This is your lambda entry point
-   */
   def handler(lambdaInput: GetCleanseListLambdaInput, context: Context): Unit = {
     val env = Env()
     logger.info(s"Starting $env")
-    logger.info(process(lambdaInput.name, env))
+    process(lambdaInput)
   }
 
-  /*
-   * I recommend to put your logic outside of the handler
-   */
-  def process(name: String, env: Env): String = s"Hello $name! (from ${env.app} in ${env.stack})\n"
+  def process(lambdaInput: GetCleanseListLambdaInput): Unit = {
+
+    val newslettersToProcess = Option(lambdaInput.newslettersToProcess) // this is set by AWS, so potentially null
```

Comment:
if the lambda receives a list of newsletters then we'll process these, otherwise we'll default on processing all the newsletters as defined by the identity team

Reply:
perhaps (later on) we should consider whether we need to add a `shouldCleanse` property to the newsletter model and only process those ones?